includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -922,6 +923,25 @@ private function validate_cdata_for_node( DOMElement $element, $cdata_spec ) {
 				return [ 'code' => self::MANDATORY_CDATA_MISSING_OR_INCORRECT ];
 			}
 		}
+
+		// When the CDATA is expected to be JSON, ensure it's valid JSON.
+		if ( 'script' === $element->tagName && in_array( $element->getAttribute( 'type' ), [ 'application/json', 'application/ld+json' ], true ) ) {
+			if ( empty( $element->textContent ) ) {
+				return [
+					'code'    => self::INVALID_JSON_CDATA,
+					'message' => esc_html__( 'Expected JSON, got an empty value', 'amp' ),
```

Comment:
Here's what this looks when the expected JSON is `empty()`:

![ms-new-error](https://user-images.githubusercontent.com/4063887/75597685-800c4400-5a5c-11ea-99f8-a27fc9cf6667.png)